src/components/calendarfunc.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from "react";
+import Calendar from "react-calendar";
+import "./Calendar.css";
+import Checklist from "./checklist";
+import MemoFunc from "./memo";
+
+import Dropdown from "react-bootstrap/Dropdown";
+
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+const CalendarFunc = () => {
+  const [selectedDate, setSelectedDate] = useState(null);
+  const [eventName, setEventName] = useState("");
+  const [events, setEvents] = useState([]);
+
+  //States for Checklist/Memo display.
+  const [showChecklist, setShowChecklist] = useState(false);
+  const [showMemo, setShowMemo] = useState(false);
+
+  // Function to select the date.
+  const handleDateSelection = (date) => {
+    setSelectedDate(date);
+  };
+
+  // Function to handle Input Change.
+  const handleEventInputChange = (event) => {
+    setEventName(event.target.value);
+  };
+
+  // Allows users to create a new event.
+  const handleEventCreation = (event) => {
+    event.preventDefault();
+    // Only accepts inputs that are not empty.
+    if (selectedDate && eventName.trim() !== "") {
+      const newEvent = {
+        name: eventName,
+        date: selectedDate,
+      };
+      setEvents([...events, newEvent]);
+      setEventName("");
+    }
+  };
+
+  // Allows users to delete a new task.
+  const handleEventDel = (index) => {
+    const updatedEvents = events.filter((_, i) => i !== index);
+    setEvents(updatedEvents);
+  };
+
+  //Countdown timer in days.
+  const DaysLeftToEvent = (eventDate) => {
+    const dateToday = new Date();
+    const dateDiff = new Date(eventDate) - dateToday;
+    //Conditions to determine whether d-day is here.
+    return dateDiff > 0 ? Math.ceil(dateDiff / (1000 * 60 * 60 * 24)) : 0;
```

Comment:
I would definitely give this calculation a variable name to improve readability!

e.g.
```
const oneDay = 1000 * 60 * 60 * 24 // ms * sec * min * h
```
or something along those lines!